### PR TITLE
Use CircleCIs pipeline.git.base_revision instead of a JSON query

### PR DESCRIPTION
**Pull request by mrmr1993**
This should circumvent the error we were seeing [here](https://app.circleci.com/pipelines/github/CodaProtocol/coda/23847/workflows/ba15bfa1-1e1a-4d39-b76f-8e669a3a6a6e/jobs/454163/parallel-runs/0/steps/0-109).

Checklist:

- [ ] Document code purpose, how to use it
  + Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  + Document test purpose, significance of failures
  + Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: scripts/compare_pr_diff_items.py [hunk 1]
```diff
@@ -12,19 +12,7 @@
...
-    if os.path.exists(pr_json):
-        os.remove(pr_json)
-    subprocess.run([
-        'curl', api_url, '-H', 'Accept: application/vnd.github.v3.json', '-o',
-        pr_json
-    ])
-    with open(pr_json, 'r') as fp:
-        json_obj = json.load(fp, encoding='utf-8')
-    base = json_obj['base']['ref']
+def run_comparison(base, compare_script):
```
Comment (psteckler) [inline scripts/compare_pr_diff_items.py:15]: can remove preceding comment

Comment (psteckler) [reply scripts/compare_pr_diff_items.py:15]: also, can rm `import json` above

File: .circleci/config.yml [hunk 2]
```diff
@@ -483,7 +487,7 @@ jobs:
             - run:
                 name: Decode Apple Certificates
                 context: org-global
-                command: bash -c '[ -z $APPLE_CERTIFICATES ] || base64 -D -o frontend/wallet/Certificates.p12 <<< $APPLE_CERTIFICATES'
+                command: bash -c '[ -z $APPLE_CERTIFICATES ] || base64 -D -o frontend/wallet/Certificates.p12 \<<< $APPLE_CERTIFICATES'
```
Comment (bkase) [inline .circleci/config.yml:490]: is the `\` intentional here?

Comment (mrmr1993) [reply .circleci/config.yml:490]: Yea, the 2.1 version of CircleCI's config uses `<<` and `>>` to delimit pipeline variables (e.g. the one added in this PR), so they need escaping when we want the normal version.

**Pull request merged**